code/datums/observation/entered.dm

```diff
@@ -13,10 +13,26 @@
 	name = "Entered"
 	expected_type = /atom
 
-/*******************
-* Entered Handling *
-*******************/
+//	Observer Pattern Implementation: Entering
+//		Registration type: /atom/movable
+//
+//		Raised when: An /atom/movable instance has entered an atom.
+//
+//		Arguments that the called proc should expect:
+//			/atom/movable/enterer: The instance that entered the atom
+//			/atom/entered: The atom that was entered
+//			/atom/old_loc: The atom the enterer came from
+//
+
+/decl/observ/entering
+	name = "Entering"
+	expected_type = /atom/movable
+
+/****************************
+* Entered/Entering Handling *
+****************************/
 
 /atom/Entered(atom/movable/enterer, atom/old_loc)
 	..()
 	events_repository.raise_event(/decl/observ/entered, src, enterer, old_loc)
+	events_repository.raise_event(/decl/observ/entering, enterer, src, old_loc)
```

Comment:
You may want to include an extra trigger in `/movable/forceMove` if the destination is `null`, for better overall accuracy and consistency with the moved event. The entered event does not do this mostly because having null as the source is technically problematic (or at any rate inefficient).

Reply:
Oh, good point. I've set a reminder to do that tomorrow.

Reply:
I actually just completely removed it, since it's redundant.